From b7bf2646db250a41984336d6321fe8f122323ad1 Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Sat, 28 Mar 2020 13:40:25 +0800
Subject: [PATCH] ARM: dts: rv1126: add thermal_zone for tsadc

add grf for tsadc node.
modify the tsadc init clk rate.
add thermal_zone node.

Change-Id: I96591dc0dcaadbb96dc08135d63207c2ca0371db
Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
---
 arch/arm/boot/dts/rv1126.dtsi | 24 ++++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/rv1126.dtsi b/arch/arm/boot/dts/rv1126.dtsi
index 1f5597b2c395..01b28eaaea25 100644
--- a/arch/arm/boot/dts/rv1126.dtsi
+++ b/arch/arm/boot/dts/rv1126.dtsi
@@ -195,6 +195,24 @@
 		};
 	};
 
+	thermal_zones: thermal-zones {
+		cpu_thermal: cpu-thermal {
+			polling-delay-passive = <20>; /* milliseconds */
+			polling-delay = <1000>; /* milliseconds */
+			sustainable-power = <977>; /* milliwatts */
+
+			thermal-sensors = <&cpu_tsadc 0>;
+		};
+
+		npu_thermal: npu-thermal {
+			polling-delay-passive = <20>; /* milliseconds */
+			polling-delay = <1000>; /* milliseconds */
+			sustainable-power = <977>; /* milliwatts */
+
+			thermal-sensors = <&npu_tsadc 0>;
+		};
+	};
+
 	timer {
 		compatible = "arm,armv7-timer";
 		interrupts = <GIC_PPI 13 (GIC_CPU_MASK_SIMPLE(4) | IRQ_TYPE_LEVEL_HIGH)>,
@@ -1014,9 +1032,10 @@
 	cpu_tsadc: tsadc@ff5f0000 {
 		compatible = "rockchip,rv1126-tsadc";
 		reg = <0xff5f0000 0x100>;
+		rockchip,grf = <&grf>;
 		interrupts = <GIC_SPI 39 IRQ_TYPE_LEVEL_HIGH>;
 		assigned-clocks = <&cru CLK_CPU_TSADC>;
-		assigned-clock-rates = <600000>;
+		assigned-clock-rates = <2000000>;
 		clocks = <&cru CLK_CPU_TSADC>, <&cru PCLK_CPU_TSADC>,
 			 <&cru CLK_CPU_TSADCPHY>;
 		clock-names = "tsadc", "apb_pclk", "phy_clk";
@@ -1031,9 +1050,10 @@
 	npu_tsadc: tsadc@ff5f8000 {
 		compatible = "rockchip,rv1126-tsadc";
 		reg = <0xff5f8000 0x100>;
+		rockchip,grf = <&grf>;
 		interrupts = <GIC_SPI 113 IRQ_TYPE_LEVEL_HIGH>;
 		assigned-clocks = <&cru CLK_NPU_TSADC>;
-		assigned-clock-rates = <600000>;
+		assigned-clock-rates = <2000000>;
 		clocks = <&cru CLK_NPU_TSADC>, <&cru PCLK_NPU_TSADC>,
 			 <&cru CLK_NPU_TSADCPHY>;
 		clock-names = "tsadc", "apb_pclk", "phy_clk";
-- 
2.35.3

